src/security/Handshake.cc

```diff
@@ -550,23 +551,26 @@ Security::HandshakeParser::ParseCertificate(const SBuf &raw, Security::CertPoint
     pCert = nullptr;
     debugs(83, 2, "TLS parsing is not supported without OpenSSL. " << raw);
 #endif
+    return pCert;
```

Comment:
If we are never return a null maybe we do not need to fix ParseCertificate to return a CertPointer.
In this case just this fix and the #ifdef inside parseServerCertificates related fixes are enough.